metrics/segmentStats.py

```diff
@@ -0,0 +1,506 @@
+                                self.tortuosityDict[self.countDict[sourceOnCycle], sourceOnCycle, point] = curveLength / curveDisplacement
+                                self.contractionDict[self.countDict[sourceOnCycle], sourceOnCycle, point] = curveDisplacement / curveLength
+                                if curveDisplacement != 0.0:
+                                    if np.log(curveDisplacement) != 0.0:
+                                        self.hausdorffDimensionDict[self.countDict[sourceOnCycle], sourceOnCycle, point] = np.log(curveLength) / np.log(curveDisplacement)
+                                self._visitedPaths.append(simplePath)
+                                self._sortedSegments.append(sortedSegment)
+                        sourceOnCycle = point
+            self.cycleInfoDict[self.cycles + nthcycle] = [len(branchPointsOnCycle), self._getLengthAndRemoveTracedPath(subGraphSkeleton, cycle, 1, remove=0)]
+        for path in self._visitedPaths:
+            self._getLengthAndRemoveTracedPath(subGraphSkeleton, simplePath, isCycle=0, remove=1)
+        self.cycles += len(cycleList)
+
+    def _tree(self, subGraphSkeleton):
+        """
+        Find statistics of a tree like structure of disjoint graph
+
+        Parameters
+        ----------
+        subGraphSkeleton: Networkx Graph
+            disjoint subgraph of the skeleton's networkx graph
+
+        cycle: list
+            list of nodes in the cycle
+
+        Returns
+        -------
+        subGraphSkeleton : Networkx graph
+            graph changed inplace with visited nodes removed
+
+        cycleInfoDict : Dictionary
+            class instance variable changed in place with key as nth cycle, value as list with
+            [number of branch points on cycle, length of the cycle]
+
+        countDict: Dictionary
+            class instance variable changed in place with key as source node,
+            value is the number of segments attached to the source node
+
+        lengthDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the length of the segment
+
+        tortuosityDict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the tortuosity of the segment (Nan but saved as zero)
+
+        contractionDictdict: Dictionary
+            class instance variable changed in place with key as index of the segment from the source node,
+            source node, target node, value is the contraction of the segment (0) since it is a cycle
+        """
+        listOfPerms = list(itertools.product(self._branchPoints, self._endPoints))
+        for sourceOnTree, item in listOfPerms:
+            if nx.has_path(subGraphSkeleton, sourceOnTree, item) and sourceOnTree != item:
```

Comment:
I think this is another case where some `continue` statements could make this more readable.